framework/db/mysql/Schema.php

```diff
@@ -295,20 +305,15 @@ protected function loadColumnSchema($info)
         $column->phpType = $this->getColumnPhpType($column);
 
         if (!$column->isPrimaryKey) {
-            /**
-             * When displayed in the INFORMATION_SCHEMA.COLUMNS table, a default CURRENT TIMESTAMP is displayed
-             * as CURRENT_TIMESTAMP up until MariaDB 10.2.2, and as current_timestamp() from MariaDB 10.2.3.
-             *
-             * See details here: https://mariadb.com/kb/en/library/now/#description
-             */
-            if (in_array($column->type, ['timestamp', 'datetime', 'date', 'time'])
-                && isset($info['default'])
-                && preg_match('/^current_timestamp(?:\(([0-9]*)\))?$/i', $info['default'], $matches)) {
-                $column->defaultValue = new Expression('CURRENT_TIMESTAMP' . (!empty($matches[1]) ? '(' . $matches[1] . ')' : ''));
-            } elseif (isset($type) && $type === 'bit') {
-                $column->defaultValue = bindec(trim(isset($info['default']) ? $info['default'] : '', 'b\''));
-            } else {
-                $column->defaultValue = $column->phpTypecast($info['default']);
+            if (isset($info['default'])) {
+                if ($this->defaultIsExpression($info)) {
+                    $column->defaultValue = new Expression($info['default']);
+                } else {
+                    $column->defaultValue = $column->phpTypecast($info['default']);
+                    if (isset($type) && $type === 'bit') {
+                        $column->defaultValue = bindec(trim(isset($info['default']) ? $info['default'] : '', 'b\''));
```

Comment:
the isset() check on `$info['default']` is now redundant. Also before, when `$info['default']` was not set, the default value was reported as 0, now it is not set. Should be changed back to previous behavior.